tools/run_tests/python_utils/dns_server.py

```diff
@@ -93,9 +99,22 @@ def start_local_dns_server_in_background(dns_server_port):
   DNSServer(resolver, port=dns_server_port, address='127.0.0.1', tcp=False).start_thread()
   DNSServer(resolver, port=dns_server_port, address='127.0.0.1', tcp=True).start_thread()
 
-if __name__ == '__main__':
-  start_local_dns_server_in_background(15353)
-  while True:
-    time.sleep(1)
-    sys.stderr.flush()
-    sys.stdout.flush()
+
+start_local_dns_server_in_background(args.dns_port)
```

Comment:
Let's put the rest of this in a `main()` function, and invoke it only `if __name__ == '__main__'`.